tc_solaris.go

```diff
@@ -0,0 +1,75 @@
+package console
+
+import (
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+func tcget(fd uintptr, p *unix.Termios) error {
+	var err error
+	p, err = unix.IoctlGetTermios(int(fd), unix.TCGETS)
+	return err
+}
+
+func tcset(fd uintptr, p *unix.Termios) error {
+	return unix.IoctlSetTermios(int(fd), unix.TCSETS, p)
+}
+
+func tcgwinsz(fd uintptr) (WinSize, error) {
+	var ws WinSize
+
+	uws, err := unix.IoctlGetWinsize(int(fd), unix.TIOCGWINSZ)
```

Comment:
we can also use this for linux as well so i dont think we need to extract these out.